### PR TITLE
[fi] Add new Ibex and OTBN tests

### PR DESCRIPTION
**Pull request by nasahlpa**
This PR contains two commits:

- Add new handlers for Ibex FI tests
- Add new handlers for OTBN FI tests

### CODE REVIEW

**Review by andrea-caforio — APPROVED** [review]

File: fault_injection/configs/pen.global_fi.ibex.char.hardened_check_eq_unimps.cw310.yaml [hunk 1]
```diff
@@ -37,6 +37,9 @@ test:
   which_test: "ibex_char_hardened_check_eq_unimp"
   # which_test: "ibex_char_hardened_check_eq_2_unimps"
   # which_test: "ibex_char_hardened_check_eq_3_unimps"
+  # which_test: "ibex_char_hardened_check_eq_4_unimps"
```
Comment (vogelpi) [inline fault_injection/configs/pen.global_fi.ibex.char.hardened_check_eq_unimps.cw310.yaml:40]: The commit message title talks about OTBN but whereas it's actually about Ibex here. This should probably be corrected.

File: fault_injection/configs/pen.global_fi.otbn.char.dmem.access.cw310.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
...46,4229950334,825123374,1038376279,2470646264,4218773437,3746395077,1448724516,3262565001,1661122272,1877452977,857694292,772569637,802843685,2125455361,3406777056,1764727446,920060882,4025369693,1349100798,2657217694,3394126751,687787461,576621678,1479550633,2357376939,1082544701,2007372129,1976310578,3156904125,592833811,3705765941,359181864,2221615505,881896932,4111468059,3200275311,2017758330,2983828617,2458693710,1318909501,761916836,3033935028,3438888195,3544373064,49464920,28038751,3663653666,762100395,1461355421,2489968999,365663631,1685236111,1515268868,3440084261,3377173686,2531354775,3364444857,355319372,3977655639,2562537247,2632720970,2322265413,4165295828,1717227322,191431321,3846398043,1421641274,1737581554,899970312,3106177760,3829243285,3798641989,1615864806,3367735900,3225082500,4154605851,3694669111,3189306406,2529953284,1801951682,1348275831,559991528,3520078732,3214041459,2057586075,2195673018,3287101809,2946758789,3811293796,2432574270,4201292549,1274171431,1009532021,1331756015,4233976805,1172338183,3572168806,2894879080,4008636142,117835012,185207048,252579084,319951120,387323156,454695192,522067228],"insn_cnt":271,"err_otbn":0,"err_ibx":0,"alerts":[0,0,0]}'
```
Comment (vogelpi) [inline fault_injection/configs/pen.global_fi.otbn.char.dmem.access.cw310.yaml:38]: Question: is the data section of the expected result here the readable part of DMEM? Are there any dependencies on RND/URND, scrambling keys or similar that could prevent successful repeating / porting of the test? Sidenote: on FPGA, the entropy source is not really random.

**Review by vogelpi — APPROVED** [review]
LGTM with one minor nit and a question.

File: fault_injection/configs/pen.global_fi.ibex.char.hardened_check_eq_unimps.cw310.yaml (hunk 1 above)
Comment (nasahlpa) [reply fault_injection/configs/pen.global_fi.ibex.char.hardened_check_eq_unimps.cw310.yaml:40]: Argh sorry, copy-and-paste error. Fixed.

File: fault_injection/configs/pen.global_fi.otbn.char.dmem.access.cw310.yaml (hunk 2 above)
Comment (nasahlpa) [reply fault_injection/configs/pen.global_fi.otbn.char.dmem.access.cw310.yaml:38]: In this test, this is the expected data. This data is defined in the DMEM section in the [test](https://github.com/nasahlpa/opentitan/blob/pentest/sw/device/tests/penetrationtests/firmware/fi/otbn/otbn_char_dmem_access.s) and it is read out by Ibex.

But when thinking about your comment, I've actually found out that one test, the CHAR.BN.WSRR test indeed fetches random data. I've added a comment in the config file and created a new issue #382.

Comment (vogelpi) [reply fault_injection/configs/pen.global_fi.otbn.char.dmem.access.cw310.yaml:38]: Thanks for the explanation, this sounds good to me!

**Pull request merged**